helix-term/src/ui/editor.rs

```diff
@@ -857,28 +857,40 @@ impl EditorView {
     }
 
     fn insert_mode(&mut self, cx: &mut commands::Context, event: KeyEvent) {
-        if let Some(keyresult) = self.handle_keymap_event(Mode::Insert, cx, event) {
-            match keyresult {
-                KeymapResult::NotFound => {
-                    if let Some(ch) = event.char() {
-                        commands::insert::insert_char(cx, ch)
+        let mut queue = VecDeque::from([event]);
```

Comment:
The VecDeque can be replaced with a regular Vec and you can just `Vec::pop`/`Vec::extend` to the end

Reply:
Good point, in fact this is a stack and not sure why I thought it was going to be a queue.